### PR TITLE
wic:mx8: Fix the image creation

### PR DESCRIPTION
**Pull request by angolini**
imx8 family boards use u-boot as boot loader in addition to some other
binaries. The final bootloader binary is gathered by mkimage into a file
called imx-boot.

Add the wks file to create the wic image tailored for imx8 family.
Fix the symbolic link file create to point to imx-boot (so it works for
all the imx8 flavors).
Set the wks file as default for all imx8.

Signed-off-by: Daiane Angolini <daiane.angolini@nxp.com>

### CODE REVIEW

File: wic/imx-imx-boot-bootpart.wks [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#  - ---------- -------------- --------------
+# ^ ^          ^              ^
+# | |          |              |
+# 0 33kiB    4MiB          16MiB + rootfs + IMAGE_EXTRA_SPACE (default 10MiB)
+#
+part u-boot --source rawcopy --sourceparams="file=imx-boot" --ondisk mmcblk --no-table --align 33
+part /boot --source bootimg-partition --ondisk mmcblk --fstype=vfat --label boot --active --align 4096 --size 16
+part / --source rootfs --ondisk mmcblk --fstype=ext4 --label root --align 4096
+
+bootloader --ptable msdos
```
Comment (otavio) [inline wic/imx-imx-boot-bootpart.wks]: Please add a newline at the end of the file.

File: recipes-bsp/u-boot/u-boot-imx_2017.03.bb [hunk 2]
```diff
@@ -35,7 +35,7 @@ do_deploy_append_mx8m() {
                     install -d ${DEPLOYDIR}/${BOOT_TOOLS}
                     install -m 0777 ${B}/${config}/arch/arm/dts/${UBOOT_DTB_NAME}  ${DEPLOYDIR}/${BOOT_TOOLS}
                     install -m 0777 ${B}/${config}/tools/mkimage  ${DEPLOYDIR}/${BOOT_TOOLS}/mkimage_uboot
-                    install -m 0777 ${B}/${config}/u-boot-nodtb.bin  ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${UBOOT_CONFIG}
+                    install -m 0777 ${B}/${config}/u-boot-nodtb.bin  ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin
```
Comment (otavio) [inline recipes-bsp/u-boot/u-boot-imx_2017.03.bb]: I am fine with this but please split this out to another patch.

File: recipes-bsp/imx-mkimage/imx-boot_0.2.bb [hunk 3]
```diff
@@ -180,7 +180,7 @@ do_deploy() {
         install -m 0644 ${S}/${BOOT_CONFIG_MACHINE}-${target} ${DEPLOYDIR}
     done
     cd ${DEPLOYDIR}
-    ln -sf ${BOOT_CONFIG_MACHINE}-${IMAGE_IMXBOOT_TARGET} ${BOOT_CONFIG_MACHINE}
+    ln -sf ${BOOT_CONFIG_MACHINE}-${IMAGE_IMXBOOT_TARGET} ${BOOT_NAME}
```
Comment (otavio) [inline recipes-bsp/imx-mkimage/imx-boot_0.2.bb]: I am fine with this but please split this out to another patch.

File: wic/imx-imx-boot-bootpart.wks (hunk 1 above)
Comment (angolini) [reply wic/imx-imx-boot-bootpart.wks]: done

File: recipes-bsp/u-boot/u-boot-imx_2017.03.bb (hunk 2 above)
Comment (angolini) [reply recipes-bsp/u-boot/u-boot-imx_2017.03.bb]: I could get it working without changing this filename

File: recipes-bsp/imx-mkimage/imx-boot_0.2.bb (hunk 3 above)
Comment (angolini) [reply recipes-bsp/imx-mkimage/imx-boot_0.2.bb]: done
